intl_translation_format/README.md

```diff
@@ -56,6 +56,169 @@ pub run intl_translation_format:generate
 ```
 
 ### Implement a new translation file format
-[TBD]
 
+The intl_translation_format brings the tools needed to support a new translation format. 
+You can extend the class TranslationFormat and complete the functions that will take care of the extracting of the messages
+and create templates. There are several classes to make this task easier: `MonoLingualFormat`, `BinaryMonolingualFormat`,
+`MultiLingualFormat`.
 
+As an example, this guide will saw you how the JSON file format is implemented.
+
+1. Learning about your format
+
+It is important to do some research about the format you are interested to implement. Some good questions would be:
+ 1. Is it a standard? 
+ 2. Is it a binary or text-based format?
+ 3. Does it contain a translation for multiple languages per file or is it just one language per file?
+ 4. How does it handle Plurals, Genders, and Variables?
+
+Let's answer these questions for our new JSON format:
+ 1. While it is a standard data interchange format, there is no official standard for translations. Multiple frameworks create their implementations like `i18next` or `ng2-translate` for `Angular`.
+
+ Because the `ICU Format` is the core of the dart intl library, we decided to create a value/key simple JSON with ICU messages. 
+
+ As an example: 
+ ```json
+ {
+   "message1" : "This is a message",
+   "message2" : "Hello { userName }, this a message with a variable",
+   "message2" : "You have { howMany, plural, =0 { none Apples} one { one Apple} other { {howMany} Apples} }"
+ } 
+ ```
+ 2. It is a text-based format
+ 3. It contains one locale translation per file.
+ 4. We will use the ICU standard for plurals, genders, and variables.
```

Comment:
I don't think ICU format is really a standard in the way you've used the term earlier. It is definitely not well enough documented to answer questions of interpretation. Maybe say "ICU format".